airbyte-cdk/python/airbyte_cdk/sources/file_based/stream/concurrent/adapters.py

```diff
@@ -45,13 +45,14 @@ def create_from_stream(
         source: AbstractSource,
         logger: logging.Logger,
         state: Optional[MutableMapping[str, Any]],
-        cursor: FileBasedNoopCursor,
+        cursor: AbstractFileBasedCursor,
     ) -> "FileBasedStreamFacade":
         """
         Create a ConcurrentStream from a FileBasedStream object.
         """
         pk = get_primary_key_from_stream(stream.primary_key)
         cursor_field = get_cursor_field_from_stream(stream)
+        stream.cursor = cursor
```

Comment:
that's unfortunate. 

looking at how the connection state manager uses the stream name to instance mapping, I think it only needs an `AirbyteStream`, not the actual stream instance so I think the path forward should be
1. Modify the state manager so it depends on `AirbyteStream`s instead of on `Stream`s
2. Instantiate the state manager before instantiating the cursor
3. Instantiate the cursor from the state manager

Reply:
@girarda that doesn't quite work either because we get the `AirbyteStream`s from the catalog, but we don't always have a catalog - e.g. during `check`.

It looks like the only pieces of information the state manager actually needs are the name and namespace, which I can get off of the config. WDYT about that modification? My worry is that we might eventually want/need other info in the `ConnectionStateManager`.

Reply:
isn't the namespace also defined in the configured_catalog?

the check operation shouldn't need the namespace or the state manager. maybe it should only be instantiated for reads